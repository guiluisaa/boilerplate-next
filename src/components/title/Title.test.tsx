import { render, screen } from '@testing-library/react';

import Title from './Title.component';

describe('component | <Title />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Title>Heading</Title>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the heading', () => {
    render(<Title>Heading</Title>);

    expect(
      screen.getByRole('heading', { name: /heading/i })
    ).toBeInTheDocument();
  });

  it('should render black text by default', () => {
    const { container } = render(<Title>Heading</Title>);

    expect(container.firstChild).toHaveStyle({ color: '#424242' });
  });

  it('should render red text when danger property is passed', () => {
    const { container } = render(<Title danger>Heading</Title>);

    expect(container.firstChild).toHaveStyle({ color: 'red' });
  });
});
