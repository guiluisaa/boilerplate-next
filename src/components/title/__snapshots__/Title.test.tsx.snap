// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | <Title /> should match snapshot 1`] = `
<h1
  class="Titlecomponent__Title-r7cffw-0 iJANgm"
>
  Heading
</h1>
`;
